rasa_nlu/evaluate.py

```diff
@@ -527,17 +516,23 @@ def run_evaluation(data_path, model_path,
     interpreter = Interpreter.load(model_path, component_builder)
     test_data = training_data.load_data(data_path,
                                         interpreter.model_metadata.language)
-
-    intent_targets = get_intent_targets(test_data)
-    entity_targets = get_entity_targets(test_data)
-    intent_predictions = get_intent_predictions(interpreter, test_data)
-    entity_predictions, tokens = get_entity_predictions(interpreter, test_data)
     extractors = get_entity_extractors(interpreter)
-    extractors, entity_predictions = patch_duckling(interpreter, extractors,
-                                                    entity_predictions)
 
-    evaluate_intents(intent_targets, intent_predictions)
-    evaluate_entities(entity_targets, entity_predictions, tokens, extractors)
+    if extractors:
+        entity_targets = get_entity_targets(test_data)
+        entity_predictions, tokens = get_entity_predictions(interpreter,
+                                                            test_data)
+        extractors, entity_predictions = remove_duckling(extractors,
```

Comment:
what happens if only duckling is in the pipeline and we remove it here - will the rest of this if block still work (given that in the beginning we check for `if extractors` which is now empty)?

Reply:
works fine, yes. Looks like this, although I'm not sure why it's printing `ner_duckling` twice:
```
2018-04-12 13:32:21 INFO     rasa_nlu.model - Starting to train component ner_duckling
2018-04-12 13:32:21 INFO     rasa_nlu.model - Finished training component.
2018-04-12 13:32:22 INFO     __main__ - Skipping evaluation of ner_duckling
2018-04-12 13:32:24 INFO     __main__ - Skipping evaluation of ner_duckling
2018-04-12 13:32:24 INFO     __main__ - CV evaluation (n=2)
2018-04-12 13:32:24 INFO     __main__ - Finished evaluation
```